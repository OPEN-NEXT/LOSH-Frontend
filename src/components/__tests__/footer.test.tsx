import * as React from "react";
import ShallowRenderer from "react-test-renderer/shallow"; // ES6
import Footer from "../footer";
import site from "../../../gatsby-config";

const renderer = ShallowRenderer.createRenderer();

jest.mock("antd", () => {
	return {
		Layout: {
			Footer: (): null => null,
		},
		Row: (): null => null,
		Typography: {
			Text: (): null => null,
		},
	};
});

describe("Footer", () => {
	it("renders correctly", () => {
		const tree = renderer.render(
			<Footer footerLinks={site.siteMetadata.footerLinks} />
		);
		expect(tree).toMatchSnapshot();
	});
});
