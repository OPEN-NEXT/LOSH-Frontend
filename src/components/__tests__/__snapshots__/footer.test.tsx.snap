// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<Footer
  style={
    Object {
      "paddingLeft": "1rem",
      "paddingRight": "1rem",
    }
  }
>
  <Row
    align="middle"
  >
    <mockConstructor
      style={
        Object {
          "paddingRight": "1rem",
          "textDecoration": "none",
        }
      }
      to="/"
    >
      <Text>
        (c) OPEN!NEXT
      </Text>
    </mockConstructor>
    <mockConstructor
      style={
        Object {
          "paddingRight": "1rem",
          "textDecoration": "none",
        }
      }
      to="https://opennext.eu/"
    >
      <Text>
        Imprint
      </Text>
    </mockConstructor>
    <mockConstructor
      style={
        Object {
          "paddingRight": "1rem",
          "textDecoration": "none",
        }
      }
      to="https://opennext.eu/"
    >
      <Text>
        Legal Disclaimer
      </Text>
    </mockConstructor>
    <mockConstructor
      style={
        Object {
          "paddingRight": "1rem",
          "textDecoration": "none",
        }
      }
      to="/issue"
    >
      <Text>
        Submit an issue
      </Text>
    </mockConstructor>
  </Row>
</Footer>
`;
